doc/bucket-replication.md

```diff
@@ -0,0 +1,97 @@
+[NooBaa Operator](../README.md) /
+
+# Bucket Replication
+Bucket replication is a NooBaa feature that allows a user to set a replication policy for all or some objects. The goal of replication policies is simple - to define a target bucket for objects to be copied to.
```

Comment:
Do we protect the user from setting the same bucket as the source and target bucket?
I think it is worth mentioning it.